engine/consensus/sealing/engine.go

```diff
@@ -438,12 +446,18 @@ func (e *Engine) OnFinalizedBlock(*model.Block) {
 // CAUTION: the input to this callback is treated as trusted; precautions should be taken that messages
 // from external nodes cannot be considered as inputs to this function
 func (e *Engine) OnBlockIncorporated(incorporatedBlock *model.Block) {
-	e.pendingIncorporatedBlocks.Push(incorporatedBlock.BlockID)
+	added := e.pendingIncorporatedBlocks.Push(incorporatedBlock.BlockID)
```

Comment:
One of the options to consider is using a buffered channel, this will allow us to avoid dropping but at cost of possibly blocking. I bet you have considered this option, would like to hear your thoughts on this.

Reply:
Completely agree, a buffered channel is a totally fine approach too. My reasons are:
* [aesthetically] we have multiple queues in the sealing engine, and therefore using also a queue here is the most consistent
* The queue is fully compliant with the API convention to _never block_:  https://github.com/onflow/flow-go/blob/f2555090b6f672036d9368f7890b8e355ccb9736/consensus/hotstuff/consumer.go#L19-L31
   * if we exceed the queue capacity, the node will go down with an error, which we will see in the logs
   * In comparison,  if we exceed the channel capacity, we block the main consensus logic. This will look like the node just hangs, without anything in the logs. I don't even want to start thinking about how painful it would be to find the root cause. I am a big fan of "fail early and fail hard (loud)" as opposed to working around an edge case that we agree should not happen in practise. 
* Lastly, I think a buffered channel allocates the entire memory upfront. Therefore, we would always pay the price for the full queue. But in practise, this queue will be empty with overwhelming probability because the worker drains it immediately, maybe it has rarely one element in it, almost never two ... Don't really think it is worth-while paying the cost for 10k elements. But we need a large capacity to avoid crashing the node too early or blocking other components of the node. 

🤷